src/spells.cpp

```diff
@@ -236,7 +236,8 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 
 InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
-	auto& it = std::advance(instants.begin(), spellId);
+	auto& it = instants.begin();
+	std::advance(it, spellId);
```

Comment:
That kills the point of the change. Since he checks if the iterator is past the end right below this line, it's fine.